packages/core-transaction-pool-mem/lib/mem.js

```diff
@@ -0,0 +1,251 @@
+      })
+
+      if (!thisIsDBLoad) {
+        if (this.dirty.removed.has(transaction.id)) {
+          // If the transaction has been already in the pool and has been removed
+          // and the removal has not propagated to disk yet, just wipe it from the
+          // list of removed transactions, so that the old copy stays on disk.
+          this.dirty.removed.delete(transaction.id)
+        } else {
+          this.dirty.added.add(transaction.id)
+        }
+      }
+    }
+  }
+
+  /**
+   * Remove a transaction.
+   * @param {String} id              id of the transaction to remove
+   * @param {String} senderPublicKey public key of the sender, could be undefined
+   */
+  remove (id, senderPublicKey) {
+    if (senderPublicKey === undefined) {
+      const transaction = this.byId.get(id)
+      if (transaction === undefined) {
+        // Not found, not in pool
+        return
+      }
+      senderPublicKey = transaction.senderPublicKey
+    }
+
+    // O(n)
+    const index = this.idsByExpiration.findIndex(function (element) {
+      return element.transactionId === id
+    })
+    if (index === -1) {
+      // Not found, not in pool
+      return
+    }
+    this.idsByExpiration.splice(index, 1)
+
+    this.idsBySender.delete(senderPublicKey)
+
+    this.byId.delete(id)
+
+    if (this.dirty.added.has(id)) {
+      // This transaction has been added and deleted without data being synced
+      // to disk in between, so it will never touch the disk, just remove it
+      // from the added list.
+      this.dirty.added.delete(id)
+    } else {
+      this.dirty.removed.add(id)
+    }
+  }
+
+  /**
+   * Get the number of transactions.
+   * @return Number
+   */
+  getSize () {
+    return this.byId.size
+  }
+
+  /**
+   * Get all transactions ids from a given sender.
+   * @param {String} senderPublicKey public key of the sender
+   * @return {Set of String} all ids for the given sender, could be empty Set
+   */
+  getIdsBySender (senderPublicKey) {
+    const ids = this.idsBySender.get(senderPublicKey)
+    if (ids !== undefined) {
+      return ids
+    }
+    return new Set()
+  }
+
+  /**
+   * Get a transaction, given its id.
+   * @param {String} id transaction id
+   * @return {Transaction|undefined}
+   */
+  getTransactionById (id) {
+    return this.byId.get(id)
+  }
+
+  /**
+   * Get an interator to all transactions in insertion order.
+   * @return {Iterator}
+   */
+  getTransactionsInInsertionOrder () {
+    return this.byId.values()
+  }
+
+  /**
+   * Check if a transaction with a given id exists.
+   * @param {String} id transaction id
+   * @return {Boolean} true if exists
+   */
+  transactionExists (id) {
+    return this.byId.has(id)
+  }
+
+  /**
+   * Get the expired transactions.
+   * @return {Array of Transaction} expired transactions
+   */
+  getExpired () {
+    const now = new Date()
+
+    let transactions = []
+
+    for (let e of this.idsByExpiration) {
+      if (e.expireAt < now) {
```

Comment:
Mhh not that it makes a big difference, but shouldn't this be semantically `less or equal`?